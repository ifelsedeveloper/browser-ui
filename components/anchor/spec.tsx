/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Anchor from '../anchor'

describe('anchor tests', () => {
  it('matches the snapshot', () => {
    const component = <Anchor href='#' text='testText' />
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component', () => {
    const wrapper = shallow(<Anchor id='anchor' href='#' text='testText' />)
    const assertion = wrapper.find('#anchor').length
    expect(assertion).toBe(1)
  })

  it('defines a href', () => {
    const wrapper = shallow(<Anchor href='https://nespresso.com' text='testText' />)
    const assertion = wrapper.props().href
    expect(assertion).toEqual('https://nespresso.com')
  })

  it('defines a target', () => {
    const wrapper = shallow(<Anchor href='#' text='testText' target='_blank' />)
    const assertion = wrapper.props().target
    expect(assertion).toEqual('_blank')
  })

  it('rel defaults to noreferrer noopener', () => {
    const wrapper = shallow(<Anchor href='#' text='testText' />)
    const assertion = wrapper.props().rel
    expect(assertion).toEqual('noreferrer noopener')
  })

  it('can pass text', () => {
    const wrapper = shallow(<Anchor href='#' text='GOTCHA' />)
    const assertion = wrapper.html().includes('GOTCHA')
    expect(assertion).toBe(true)
  })
})
