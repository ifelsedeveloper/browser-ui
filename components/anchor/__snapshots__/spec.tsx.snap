// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anchor tests matches the snapshot 1`] = `
<a
  className="sc-bdVaJa gNSFuh"
  href="#"
  id={undefined}
  rel="noreferrer noopener"
  target={undefined}
>
  testText
</a>
`;
